contracts/mainnet/connectors/mstable/main.sol

```diff
@@ -0,0 +1,367 @@
+
+		_eventName = "LogClaimRewards(address,uint256)";
+		_eventParam = abi.encode(rewardToken, claimedRewardToken);
+	}
+
+	/**
+	 * @dev Swap tokens
+	 * @notice Swaps tokens via Masset basket
+	 * @param _input Token address to swap from
+	 * @param _output Token address to swap to
+	 * @param _amount Amount of tokens to swap
+	 * @param _minOut Minimum amount of token to mint
+	 * @return _eventName Event name
+	 * @return _eventParam Event parameters
+	 */
+
+	function swap(
+		address _input,
+		address _output,
+		uint256 _amount,
+		uint256 _minOut
+	) external returns (string memory _eventName, bytes memory _eventParam) {
+		//
+		approve(TokenInterface(_input), mUsdToken, _amount);
+		uint256 amountSwapped;
+
+		// Check the assets and swap accordingly
+		if (_output == mUsdToken) {
+			// bAsset to mUSD => mint
+			amountSwapped = IMasset(mUsdToken).mint(
+				_input,
+				_amount,
+				_minOut,
+				address(this)
+			);
+		} else if (_input == mUsdToken) {
+			// mUSD to bAsset => redeem
+			amountSwapped = IMasset(mUsdToken).redeem(
+				_output,
+				_amount,
+				_minOut,
+				address(this)
+			);
+		} else {
+			// bAsset to another bAsset => swap
+			amountSwapped = IMasset(mUsdToken).swap(
+				_input,
+				_output,
+				_amount,
+				_minOut,
+				address(this)
+			);
+		}
+
+		_eventName = "LogSwap(address,address,uint256,uint256)";
+		_eventParam = abi.encode(_input, _output, _amount, amountSwapped);
+	}
+
+	/**
+	 * @dev Swap tokens via Feeder Pool
+	 * @notice Swaps tokens via Feeder Pool
+	 * @param _input Token address to swap from
+	 * @param _output Token address to swap to
+	 * @param _amount Amount of tokens to swap
+	 * @param _minOut Minimum amount of token to mint
+	 * @param _path Feeder Pool address to use
+	 * @return _eventName Event name
+	 * @return _eventParam Event parameters
+	 */
+
+	function swapViaFeeder(
+		address _input,
+		address _output,
+		uint256 _amount,
+		uint256 _minOut,
+		address _path
+	) external returns (string memory _eventName, bytes memory _eventParam) {
+		//
+		uint256 amountSwapped;
+
+		approve(TokenInterface(_input), _path, _amount);
+
+		// swaps fAsset to mUSD via Feeder Pool
+		// swaps mUSD to fAsset via Feeder Pool
+		amountSwapped = IFeederPool(_path).swap(
+			_input,
+			_output,
+			_amount,
+			_minOut,
+			address(this)
+		);
+
+		_eventName = "LogSwap(address,address,uint256,uint256)";
+		_eventParam = abi.encode(_input, _output, _amount, amountSwapped);
+	}
+
+	/***************************************
+                    Internal
+    ****************************************/
+
+	/**
+	 * @dev Deposit to Save from any asset
+	 * @notice Called internally from deposit functions
+	 * @param _token Address of token to deposit
+	 * @param _amount Amount of token to deposit
+	 * @param _path Path to mint mUSD (only needed for Feeder Pool)
+	 * @return _eventName Event name
+	 * @return _eventParam Event parameters
+	 */
+
+	function _deposit(
```

Comment:
Why don't we move `_deposit`, `_withdraw`, `_getRewardToken`, `_getRewardInternalBal ` this functions to `helpers.sol` ?